state-chain/pallets/cf-pools/src/lib.rs

```diff
@@ -472,6 +532,26 @@ pub mod pallet {
 					}
 				}
 			}
+			let scheduled_limit_orders = ScheduledLimitOrders::<T>::take(current_block);
+			for order_update in scheduled_limit_orders {
```

Comment:
Use `for OrderUpdate { lp, id, call } ` and you can avoid some unnecessary clones.